assemblyline_v4_service/updater/helper.py

```diff
@@ -188,7 +192,13 @@ def git_clone_repo(source: Dict[str, Any], previous_update: int = None, logger=N
     ignore_ssl_errors = source.get("ssl_ignore_errors", False)
     ca_cert = source.get("ca_cert")
     proxy = source.get('proxy', None)
-    auth = f'{username}:{password}@' if username and password else None
+    auth = None
+    if username and password:
+        # Basic authentication scheme
+        auth = f'{username}:{password}@'
+    elif password:
+        # Token-based authentication
+        auth = f'{password}@'
```

Comment:
I think we could use the following in case someone wants to only use a username, for some reasons? (or if they put the token in the username?)
```suggestion
    elif username or password:
        # Token-based authentication
        auth = f'{(username or password)}@'
```

Reply:
Well if they put the token in the username, then that's exposing a secret which is what the password field is for.

Not sure if there is a use-case of providing a username without a password

Reply:
With the hidden field type for password, it makes more sense to only use the password for the token!